CRM/Utils/Array.php

```diff
@@ -60,16 +60,7 @@ public static function cast($value) {
    *   Can return any type, since $list might contain anything.
    */
   public static function value($key, $list, $default = NULL) {
-    if (is_array($list)) {
-      return array_key_exists($key, $list) ? $list[$key] : $default;
-    }
-    if ($list instanceof ArrayAccess) {
-      // ArrayAccess requires offsetExists is implemented for the equivalent to array_key_exists.
-      return $list->offsetExists($key) ? $list[$key] : $default;
-    }
-    // @todo - eliminate these from core & uncomment this line.
-    // CRM_Core_Error::deprecatedFunctionWarning('You have passed an invalid parameter for the "list"');
```

Comment:
Maybe as a first pass it would be interesting to uncomment this line and see what tests fail...

Reply:
@coleman I agree it would be interesting to know which callers are passing in non-arrays in the `$list` param, although it doesn't get at the [existing difference](https://github.com/civicrm/civicrm-core/pull/27830) between this function and `??`. Uncommenting this line would actually make this function _more strict_ than `??`. I wonder if there's reason to do that?